### ISSUE TITLE
Are you interested in adding MPO

### ISSUE THREAD

**Issue by Jogima-cyber** [vwxyzjn/cleanrl#390]
Hi, the MPO algorithm, first published in 2018, has become, it seems the preferred deepmind algorithm.

They published [2018a](https://arxiv.org/abs/1806.06920) introducing it and then several variations and improvements : [2018b](https://arxiv.org/pdf/1812.02256.pdf), [2020](https://arxiv.org/pdf/2005.07513.pdf), [2022](https://arxiv.org/abs/2204.10256). 

Recently in [2023](https://arxiv.org/pdf/2304.13653.pdf) they applied it to real robots to play soccer.

There are several deepmind's official implementations : [tensorflow implementation](https://github.com/deepmind/acme/tree/master/acme/agents/tf/mpo), [jax implementation](https://github.com/deepmind/acme/tree/master/acme/agents/jax/mpo) as well as an [example](https://github.com/deepmind/acme/tree/master/examples/baselines/rl_continuous) setting some hyperparameters different from the default ones in the implementation to "match the published results".

The deepmind's implementations are modular and rely on several of their own libraries, making it difficult to understand the code. Therefore I think it might be a good idea to add it to clean RL. Other open versions I've come across didn't get it quite right.

I've re-implemented it in torch (the [2018b version](https://arxiv.org/pdf/1812.02256.pdf)) in a minimal one-file implementation in the style of clean RL and I can match the published results with it so I wanted to know if you would be interested in adding it to clean RL? And also in benchmarking it?

**Comment by vwxyzjn** [comment]
@Jogima-cyber, thanks for raising the issue. MPO is an interesting algorithm that I would love to learn more about. It would be great to have a CleanRL MPO implementation and its associated benchmarks.


I have some quick questions.

1. Is MPO preferred by Deepmind mainly for continuous control? I noticed that in the [Muesli paper](https://arxiv.org/pdf/2104.06159.pdf), it is shown that MPO even underperforms vanilla policy gradient. 
<img width="700" alt="image" src="https://github.com/vwxyzjn/cleanrl/assets/5555347/8f926ba3-29af-49d1-acea-eb3d62845973">

2. >  I can match the published results with it
Very nice! Are you referring to the published results in DMC environments? We also have results on [DMC environments with PPO](https://docs.cleanrl.dev/rl-algorithms/ppo/#experiment-results_2), so it would be quite interesting to compare :)

**Comment by Jogima-cyber** [comment]
To answer you two questions : 

1. Sorry, I wasn't precise enough, I meant robotics continuous control. For other continuous control tasks they made a close source online version of MPO called [V-MPO](https://arxiv.org/abs/1909.12238)
2. I've only tested it on Hopper-v4 for now and I can match (even outperform, but I've used only 1 seed) the results presented in the [introduction paper, 2018a](https://arxiv.org/abs/1806.06920)

Should I make a pull request with my MPO torch implementation?

**Comment by vwxyzjn** [comment]
I see. That's good to know. A PR with MPO torch implementation sounds good. We'd probably want to run benchmark experiments in 24 envs with three random seeds, such as in Figure 4. The contribution guide can be found here https://docs.cleanrl.dev/contribution/.

**Comment by Jogima-cyber** [comment]
Thank you, I'm on it. Expect to hear from me soon regarding this then (on discord I think). I'm closing the issue.
